docs/user/azure/install_upi.md

```diff
@@ -0,0 +1,559 @@
+example.
+
+## Prerequisites
+
+* all prerequisites from [README](README.md)
+* the following binaries installed and in $PATH:
+  * [openshift-install][openshiftinstall]
+  * [az (Azure CLI)][azurecli] installed and aunthenticated
+  * python3
+  * [jq][jqjson]
+  * [yq][yqyaml]
+* python dotmap library: install it with `pip install dotmap`
+
+## Create an install config
+
+Create an install configuration as for [the usual approach](install.md#create-configuration):
+
+```console
+$ openshift-install create install-config
+? SSH Public Key /home/user_id/.ssh/id_rsa.pub
+? Platform azure
+? azure subscription id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+? azure tenant id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+? azure service principal client id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+? azure service principal client secret xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+INFO Saving user credentials to "/home/user_id/.azure/osServicePrincipal.json"
+? Region centralus
+? Base Domain example.com
+? Cluster Name test
+? Pull Secret [? for help]
+```
+
+### Extract data from install config
+
+Some data from the install configuration file will be used on later steps. Export them as environment variables with:
+
+```sh
+export CLUSTER_NAME=`yq -r .metadata.name install-config.yaml`
+export AZURE_REGION=`yq -r .platform.azure.region install-config.yaml`
+export SSH_KEY=`yq -r .sshKey install-config.yaml | xargs`
+export BASE_DOMAIN=`yq -r .baseDomain install-config.yaml`
+export BASE_DOMAIN_RESOURCE_GROUP=`yq -r .platform.azure.baseDomainResourceGroupName install-config.yaml`
+export RESOURCE_GROUP=$CLUSTER_NAME
+```
+
+### Empty the compute pool
+
+We'll be providing the compute machines ourselves, so edit the resulting `install-config.yaml` to set `replicas` to 0 for the `compute` pool:
+
+```sh
+python -c '
+import yaml;
+path = "install-config.yaml";
+data = yaml.full_load(open(path));
+data["compute"][0]["replicas"] = 0;
+open(path, "w").write(yaml.dump(data, default_flow_style=False))'
+```
+
+## Create manifests
+
+Create manifests to enable customizations that are not exposed via the install configuration.
+
+```console
+$ openshift-install create manifests
+INFO Credentials loaded from file "/home/user_id/.azure/osServicePrincipal.json"
+INFO Consuming "Install Config" from target directory
+WARNING Making control-plane schedulable by setting MastersSchedulable to true for Scheduler cluster settings
+```
+
+### Customize manifests
+
+The manifests need to reflect the resources to be created by the [Azure Resource Manager][azuretemplates] template, e.g. the
+resource group name. Also, we don't want [the ingress operator][ingress-operator] to create DNS records (we're going to
+do it manually) so we need to remove the `privateZone` and `publicZone` sections from the DNS configuration in manifests.
+
+A Python script named `setup-manifests.py` is provided to help with these and a few other changes required in manifests. This script
```

Comment:
I think we should do  include high level steps of what this `setup-manifests.py` does to show what steps are being done..